src/pages/ServiceDetails/ServiceInfo/ServiceInfoPods.tsx

```diff
@@ -1,66 +1,97 @@
 import * as React from 'react';
-import { Col, Row } from 'patternfly-react';
-import LocalTime from '../../../components/Time/LocalTime';
+import { Col, Row, OverlayTrigger, Tooltip } from 'patternfly-react';
 import Badge from '../../../components/Badge/Badge';
 import { Pod } from '../../../types/ServiceInfo';
 import { PfColors } from '../../../components/Pf/PfColors';
+import { PodsGroup, groupPods } from './ServiceInfoPodsGrouping';
 
 interface Props {
   pods?: Pod[];
 }
 
 class ServiceInfoPods extends React.Component<Props> {
+  groups: PodsGroup[];
```

Comment:
Just curiosity. Nothing to change, something to learn.
I see several styles for adding "state" into the component.
Instead to use the proper React state, here we are using a plain js object (which is right).
Why don't use the React state ? Just thinking loud if using this could have some undesired side effect through the React lifecycle. Probably not, it seems quite simple (and correct) use case, but just wondering what is the best practice to use.
I see you use it in several places on the code, and in others people use the react state, which also looks good.

Reply:
Personally I tend to reduce the react state at its minimum, that is, it only contains data that may change through the component's life.

Here `PodsGroup` is directly derived from props and will not change unless props change. So in my understanding it's not a good candidate for state.

There's other related concepts that I don't know very well yet, such as "PureComponents" and "SFC", maybe I should use it there but I'm not sure.